exercises/verify.cpp

```diff
@@ -9,10 +9,29 @@ The number 3 is [not] present in the array.
 #include <iostream>
 using namespace std;
 
+bool check(int array[], int dim, int number)
+{
+  for (int i = 0; i < dim; i++)
+  {
+    if (array[i] == number)
+    {
+      return true;
+    }
+  }
+  return false;
+}
+
 int main()
 {
   // placeholder
-  int N[10] = [ 3, 4, 5, 1, 2, 3, 4, 9, 13, 0 ];
+  int DIM = 10;
```

Comment:
questo potrebbe essere un #define, risparmi memoria ed è più professionale

Reply:
Meglio un #define o una costante?

Reply:
> Meglio un #define o una costante?

Lo chiedo perché un define se non erro può essere sovrascritto nel codice successivamente, cosa che le costanti dovrebbero vietare

Reply:
meglio #define, volendo esiste `constexpr` che se non erro funziona un po' come define solo che puoi definire il "tipo", ma non vorrei sbagliarmi.

tieni in mente che `#define` non ti ruba memoria, non crei alcuna "variabile" all'interno del codice, le "variabili define" a "compile time" vengono sostituite con quello che hai scritto

quindi scrivere int[10] o int[DIM] dove DIM è un define di valore 10 è esattamente la stessa cosa dopo che compili il codice, il benefit sta nel racchiudere quel 10 in una "finta variabile" che esiste solo nel codice, ma dopo la compilazione non esiste.

creare una variabile `const` invece creerebbe dello spazio in memoria inutile, perchè anche dopo la compilazione la vairabile const si mantiene in memoria.